academy/ibc-dev/ibcmw-integrate.md

```diff
@@ -0,0 +1,188 @@
+      transfer.NewAppModule(transferKeeper),
+      custom.NewAppModule(customKeeper)
+  )
+
+  ```
+
+    </ExpansionPanel>
+
+- As mentioned in the previous section, if the middleware wishes to send a packet or acknowledgment without the involvement of the underlying application, it should be given access to the same `scopedKeeper` as the base application so that it can retrieve the capabilities by itself.
+
+    <ExpansionPanel title="Example code">
+
+  ```go
+      //app.go
+
+      // add scoped keepers in case the middleware wishes to
+      // send a packet or acknowledgment without
+      // the involvement of the underlying application
+      scopedKeeperTransfer := capabilityKeeper.NewScopedKeeper("transfer")
+      scopedKeeperCustom1 := capabilityKeeper.NewScopedKeeper("custom1")
+      scopedKeeperCustom2 := capabilityKeeper.NewScopedKeeper("custom2")
+
+
+  ```
+
+    </ExpansionPanel>
+
+  For example, if the middleware `mw1` needs the ability to send a packet on custom2's port without involving the underlying application `custom2`, it would require access to the latter's `scopedKeeper`:
+
+  ```diff
+  - mw1Keeper := mw1.NewKeeper(storeKey1)
+  + mw1Keeper := mw1.NewKeeper(storeKey1, scopedKeeperCustom2)
+  ```
+
+<HighlightBox type="note">
+
+If no middleware (or other modules for that matter) need access to the `scopedKeeper`, there is no need to define them.
+
+</HighlightBox>
+
+- Each application stack must reserve its own unique port with core IBC. Thus two stacks with the same base application must bind to separate ports.
+
+    <ExpansionPanel title="Example code">
+
+  ```go
+      // initialize base IBC applications
+      //
+      // if you want to create two different stacks with the same base application,
+      // they must be given different scopedKeepers and assigned different ports
+      transferIBCModule := transfer.NewIBCModule(transferKeeper)
+      customIBCModule1 := custom.NewIBCModule(customKeeper, "portCustom1")
+      customIBCModule2 := custom.NewIBCModule(customKeeper, "portCustom2")
+
+  ```
+
+    </ExpansionPanel>
+
+- **The order of middleware matters**. Function calls from IBC to the application travel from top-level middleware through to bottom-level middleware and then to the application. Function calls from the application to IBC rise from the bottom middleware in order to the top middleware and then to core IBC handlers. Thus the same set of middleware arranged in different orders may produce different effects.
+
+  <HighlightBox type="tip">
+
+  In addition to the diagram shown in the [introduction](./ibcmw-intro.md), also note the direction of information flow through the middleware by looking at the interface:
+
+  ```go
+      type Middleware interface {
+          IBCModule
+          ICS4Wrapper
+  }
+  ```
+
+  The packet and channel callbacks defined by `IBCModule` run from IBC Core to the base application; the methods defined by `ICS4Wrapper` run from base application to core IBC.
```

Comment:
@tmsdkeys "from base application to core IBC" or "from base application to IBC Core". I went for the latter, if this isn't correct please change back if changes are already committed.